analyzer/src/test/kotlin/PackageJsonUtilTest.kt

```diff
@@ -0,0 +1,178 @@
+
+            hasYarnLockFile("a") shouldBe true
+        }
+
+        "given project with yarn and npm lockfile, definition file is mapped for yarn only" {
+            setupProject(path = "a", hasNpmLockFile = true, hasYarnLockFile = true)
+
+            mapDefinitionFilesForNpm(definitionFiles).shouldBeEmpty()
+            mapDefinitionFilesForYarn(definitionFiles) shouldContainExactly
+                    absolutePaths(
+                    "a/package.json")
+        }
+
+        "given project with no lockfile, definition file is mapped for npm only" {
+            setupProject(path = "a")
+
+            mapDefinitionFilesForNpm(definitionFiles) shouldContainExactly
+                    absolutePaths("a/package.json")
+            mapDefinitionFilesForYarn(definitionFiles).shouldBeEmpty()
+        }
+
+        "given project matched exactly by workspace, matched project is not mapped" {
+            setupProject(path = "a", matchers = listOf("b"))
+            setupProject(path = "a/b")
+            setupProject(path = "a/c")
+
+            mapDefinitionFiles(definitionFiles) shouldContainExactlyInAnyOrder
+                    absolutePaths(
+                            "a/package.json",
+                            "a/c/package.json")
+        }
+
+        "given projects matched via * by workspace, matched projects are not mapped" {
+            setupProject(path = "a", matchers = listOf("*", "*/f"))
+            setupProject(path = "a/b")
+            setupProject(path = "a/c")
+            setupProject(path = "a/d/e")
+            setupProject(path = "a/d/f")
+
+            mapDefinitionFiles(definitionFiles) shouldContainExactlyInAnyOrder
+                    absolutePaths(
+                            "a/package.json",
+                            "a/d/e/package.json")
+        }
+
+        "given projects matched via ** by workspace, matched projects are not mapped" {
+            setupProject(path = "a", matchers = listOf("**/d"))
+            setupProject(path = "a/b/c/d")
+            setupProject(path = "a/b/c/e")
+
+            mapDefinitionFiles(definitionFiles) shouldContainExactlyInAnyOrder
+                    absolutePaths(
+                            "a/package.json",
+                            "a/b/c/e/package.json")
+        }
+    }
+
+    private lateinit var tempDir: File
+    private val definitionFiles = mutableSetOf<File>()
+
+    override fun beforeTest(description: Description) {
+        super.beforeTest(description)
+        tempDir = createTempDir()
+        definitionFiles.clear()
+    }
+
+    override fun afterTest(description: Description, result: TestResult) {
+        tempDir.safeDeleteRecursively()
+        definitionFiles.clear()
+        super.afterTest(description, result)
+    }
+
+    private fun setupProject(path: String, matchers: List<String> = emptyList(),
+                             hasNpmLockFile: Boolean = false, hasYarnLockFile: Boolean = false) {
```

Comment:
Use the syntax from the Kotlin coding conventions (yes, we don't do this consistently, but I'd like to change that):
https://kotlinlang.org/docs/reference/coding-conventions.html#function-formatting